### ISSUE TITLE
XACQUIRE/XRELEASE and lock directive is not supported

### ISSUE THREAD

**Issue by harm-smits** [dude719/JetBrains-NASM-Language#22]
When implementing spinlocks I came accross this weird bug where the syntax highlighting for `Hardware Lock Elision Prefix Hints` doesnt work. Furthermore, it also seems like the `lock` prefix for e.g. `xchg` doesnt work.

**Comment by ajkhoury** [comment]
I'll have to add the HLE locking instruction prefixes to the language regex, thanks for reporting this!

**Comment by ajkhoury** [comment]
@harm-smits Can you post a snippet of the HLE instructions that are not highlighted properly?

**Comment by harm-smits** [comment]
@ajkhoury Here you go:

```asm

%include "macros.asm"

section .text
	global LABEL(ft_spin_lock)

LABEL(ft_spin_lock):
	mov		rcx, 1

.retry:
	xor		rax, rax
	xacquire lock cmpxchg BYTE [rdi], cl
	je		.out

.pause:
	mov		rax, [rdi]
	test	rax, rax
	je		.retry
	rep		nop
	jmp		.pause

.out:
	ret

```

**Comment by harm-smits** [comment]
Could also give you access to my library repo, there is a bunch of ASM code in there that you could use for debugging stuff if you want.

**Comment by ajkhoury** [comment]
Fixed in 322f7a7
